pkg/apis/build/v1alpha2/build_pod_test.go

```diff
@@ -2703,6 +2718,224 @@ func testBuildPod(t *testing.T, when spec.G, it spec.S) {
+				validateNoPrivilegeEscalation(rebasePod.Spec.Containers)
+			})
+			it("only uses allowed app armor values", func() {
+				validateAppArmor := func(pod *corev1.Pod) {
+					for key, value := range pod.Annotations {
+						if strings.HasPrefix(key, corev1.AppArmorBetaContainerAnnotationKeyPrefix) {
+							or(assert.Equal(t, corev1.AppArmorBetaProfileNamePrefix, value), assert.Equal(t, corev1.AppArmorBetaProfileRuntimeDefault, value))
+						}
+					}
+				}
+
+				validateAppArmor(pod)
+				validateAppArmor(rebasePod)
+			})
+			it("only uses allowed se linux types", func() {
+				validateSELinux := func(options *corev1.SELinuxOptions) {
+					if options != nil {
+						or(assert.Equal(t, "", options.Type),
+							assert.Equal(t, "container_t", options.Type),
+							assert.Equal(t, "container_init_t", options.Type),
+							assert.Equal(t, "container_kvm_t", options.Type))
+
+						assert.Equal(t, "", options.User)
+						assert.Equal(t, "", options.Role)
+					}
+				}
+
+				validateContainerSELinux := func(containers []corev1.Container) {
+					for _, container := range containers {
+						validateSELinux(container.SecurityContext.SELinuxOptions)
+					}
+				}
+
+				validateSELinux(pod.Spec.SecurityContext.SELinuxOptions)
+				validateSELinux(rebasePod.Spec.SecurityContext.SELinuxOptions)
+				validateContainerSELinux(pod.Spec.InitContainers)
+				validateContainerSELinux(pod.Spec.Containers)
+				validateContainerSELinux(rebasePod.Spec.InitContainers)
+				validateContainerSELinux(rebasePod.Spec.Containers)
+			})
+			it("uses allowed volume types", func() {
+				validateCorrectVolumeTypes := func(volumes []corev1.Volume) {
+					for _, volume := range volumes {
+						if volume.ConfigMap == nil &&
+							volume.CSI == nil &&
+							volume.DownwardAPI == nil &&
+							volume.EmptyDir == nil &&
+							volume.Ephemeral == nil &&
+							volume.PersistentVolumeClaim == nil &&
+							volume.Projected == nil &&
+							volume.Secret == nil {
+							assert.Fail(t, "invalid volume spec ")
+						}
+						assert.Nil(t, volume.HostPath)
+					}
+				}
+
+				validateCorrectVolumeTypes(pod.Spec.Volumes)
+				validateCorrectVolumeTypes(rebasePod.Spec.Volumes)
+
+			})
+			it("uses allowed proc mounts", func() {
+				validateProcMount := func(containers []corev1.Container) {
+					for _, container := range containers {
+						if procMount := container.SecurityContext.ProcMount != nil; procMount {
+							assert.Equal(t, corev1.DefaultProcMount, procMount)
+						}
+					}
+				}
+
+				validateProcMount(pod.Spec.InitContainers)
+				validateProcMount(pod.Spec.Containers)
+				validateProcMount(rebasePod.Spec.InitContainers)
+				validateProcMount(rebasePod.Spec.Containers)
+			})
+			it("uses allowed sysctl", func() {
+				validateProcMount := func(pod *corev1.Pod) {
+					for _, sysctl := range pod.Spec.SecurityContext.Sysctls {
+						or(assert.Equal(t, "kernel.shm_rmid_forced", sysctl.Name),
```

Comment:
What is the or doing here?

Reply:
I was trying to assert that multiple values are valid

Reply:
but now that i think of it, any call to assert that fails will result in the error being set in t so this may not work

Reply:
At this point can we validate that restricted pod security standard is met by just checking that no Sysctls are set?

Reply:
true, but i was hoping to validate the full spec so that if we make changes it will cause the test to fail

Reply:
we can do that and in the future if we need to use sysctl we can expand the test to match the actual spec

Reply:
I just asserted that those fields are unset for now